### PR TITLE
Switch to ruff linter and formatter

### PR DESCRIPTION
**Pull request by danielhollas**
I've taken ruff config from AWB (with minor modifications as needed).

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@ updates:
     - package-ecosystem: github-actions
       directory: /
       schedule:
-          interval: daily
+          interval: monthly
```
Comment (danielhollas) [inline .github/dependabot.yml:8]: This seems a bit saner and matches what we have in AWB.

File: aiidalab/app.py [hunk 2]
```diff
@@ -569,15 +569,15 @@ def install(
                         "Performing rollback to previously installed version."
                     )
                     self._restore_from(trash_path)
-            except RuntimeError as inner_error:
-                logger.error(f"Rollback failed due to error: {inner_error}!")
+            except RuntimeError:
+                logger.exception("Rollback failed! Consider re-installing the app.")
```
Comment (danielhollas) [inline aiidalab/app.py:573]: Apparently, the exception is always automatically included when using `logger.exception` inside except statement.

File: aiidalab/registry/web.py [hunk 3]
```diff
@@ -121,7 +122,6 @@ def build(
                 api_path=base_path / api_path,
                 apps_index=apps_index,
                 apps_data=apps_data,
-                scan_app_repository=parse_app_repo,
```
Comment (danielhollas) [inline aiidalab/registry/web.py:124]: This argument was completely unused in `build_api_v1`, perhaps copy-paste error or some historical remnant?

**Review by yakutovicha — APPROVED** [review]
LGTM!

**Pull request merged**